csharp/src/Ice/IncomingRequestFrame.cs

```diff
@@ -36,23 +36,23 @@ public IncomingRequestFrame(Protocol protocol, ArraySegment<byte> payload, int s
             Facet = istr.ReadFacet();
             Operation = istr.ReadString();
             IsIdempotent = istr.ReadOperationMode() != OperationMode.Normal;
-            Context = istr.ReadContext();
-            Encapsulation = payload.Slice(istr.Pos);
+            Context = Protocol == Protocol.Ice1 ? istr.ReadContext().ToImmutableDictionary() :
```

Comment:
I don't see why we're converting the dictionary to an immutable dictionary here. And why this assigment to ImmutableDictionary<string, string>.Empty when set the actual value a few lines below?

Reply:
`ToImmutableDictionary` is not required, we need to assign it here to avoid uninitialized field warnings, because the assignment below is inside a conditional the compiler doesn't know the field is always initialized

Reply:
Fixed